File: src/main/java/tw/racket/lang/RacketColorSettingsPage.java
package tw.racket.lang;

import com.intellij.openapi.editor.colors.TextAttributesKey;
import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.options.colors.AttributesDescriptor;
import com.intellij.openapi.options.colors.ColorDescriptor;
import com.intellij.openapi.options.colors.ColorSettingsPage;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.util.Map;

public class RacketColorSettingsPage implements ColorSettingsPage {
    @Nullable
    @Override
    public Icon getIcon() {
        return null;
    }

    @NotNull
    @Override
    public SyntaxHighlighter getHighlighter() {
        return new RacketSyntaxHighlighter();
    }

    @NotNull
    @Override
    public String getDemoText() {
        return "TODO";
    }

    @Nullable
    @Override
    public Map<String, TextAttributesKey> getAdditionalHighlightingTagToDescriptorMap() {
        return null;
    }

    @NotNull
    @Override
    public AttributesDescriptor[] getAttributeDescriptors() {
        return new AttributesDescriptor[]{
                new AttributesDescriptor("String", RacketSyntaxHighlighter.STRING),
                new AttributesDescriptor("Number", RacketSyntaxHighlighter.NUMBER),
                new AttributesDescriptor("Comment", RacketSyntaxHighlighter.COMMENT),
                new AttributesDescriptor("Braces", RacketSyntaxHighlighter.BRACES),
                new AttributesDescriptor("Parentheses", RacketSyntaxHighlighter.PARENTHESES),
                new AttributesDescriptor("Bad Value", RacketSyntaxHighlighter.BAD_CHARACTER),
        };
    }

    @NotNull
    @Override
    public ColorDescriptor[] getColorDescriptors() {
        return ColorDescriptor.EMPTY_ARRAY;
    }

    @Nls(capitalization = Nls.Capitalization.Title)
    @NotNull
    @Override
    public String getDisplayName() {
        return "Racket";
    }
}

File: src/main/kotlin/tw/racket/lang/RacketColorSettingsPage.kt
package tw.racket.lang

import com.intellij.openapi.editor.DefaultLanguageHighlighterColors
import com.intellij.openapi.editor.HighlighterColors
import com.intellij.openapi.editor.colors.TextAttributesKey
import com.intellij.openapi.fileTypes.SyntaxHighlighter
import com.intellij.openapi.options.colors.AttributesDescriptor
import com.intellij.openapi.options.colors.ColorDescriptor
import com.intellij.openapi.options.colors.ColorSettingsPage
import org.jetbrains.annotations.Nls
import javax.swing.Icon

class RacketColorSettingsPage : ColorSettingsPage {
    override fun getIcon(): Icon? {
        return null
    }

    override fun getHighlighter(): SyntaxHighlighter {
        return RacketSyntaxHighlighter()
    }

    override fun getDemoText(): String {
        return "TODO"
    }

    override fun getAdditionalHighlightingTagToDescriptorMap(): Map<String, TextAttributesKey>? {
        return null
    }

    override fun getAttributeDescriptors(): Array<AttributesDescriptor> {
        return arrayOf(
                AttributesDescriptor("String", TextAttributesKey.createTextAttributesKey("RACKET_STRING", DefaultLanguageHighlighterColors.STRING)),
                AttributesDescriptor("Number", TextAttributesKey.createTextAttributesKey("RACKET_NUMBER", DefaultLanguageHighlighterColors.NUMBER)),
                AttributesDescriptor("Comment", TextAttributesKey.createTextAttributesKey("RACKET_COMMENT", DefaultLanguageHighlighterColors.LINE_COMMENT)),
                AttributesDescriptor("Braces", TextAttributesKey.createTextAttributesKey("RACKET_BRACES", DefaultLanguageHighlighterColors.BRACES)),
                AttributesDescriptor("Parentheses", TextAttributesKey.createTextAttributesKey("RACKET_PARENTHESES", DefaultLanguageHighlighterColors.PARENTHESES)),
                AttributesDescriptor("Bad Value", TextAttributesKey.createTextAttributesKey("RACKET_BAD_CHARACTER", HighlighterColors.BAD_CHARACTER)))
    }

    override fun getColorDescriptors(): Array<ColorDescriptor> {
        return ColorDescriptor.EMPTY_ARRAY
    }

    @Nls(capitalization = Nls.Capitalization.Title)
    override fun getDisplayName(): String {
        return "Racket"
    }
}